packages/babel-preset-gatsby/src/index.js

```diff
@@ -119,12 +119,19 @@ export default function preset(_, options = {}) {
           // absoluteRuntime: absoluteRuntimePath,
         },
       ],
+      // TODO allow loose mode as an option in v3
       isBrowser && [
         resolve(`@babel/plugin-transform-spread`),
         {
           loose: false, // Fixes #14848
         },
       ],
+      isBrowser && [
```

Comment:
Instead of transpiling both class and spread unconditionally, consider check if `transform-spread` is actually required for the given targets:

```js
import { isRequired } from "@babel/helper-compilation-targets";

isRequired("transform-spread", targets)
// returns boolean, whether `transform-spread` is `required`.
```

You can plugin transform-classes safely because if a target does not support `...` natively, it does not support class either.

Reply:
TIL `@babel/helper-compilation-targets`. Thanks!

Reply:
@babel/helper-compilation-targets didn't accept targets how we handle it and pass it on to preset-env so I have to do some more digging and make it more resilient. I've made a ticket in our issue tracker

Reply:
`@babel/preset-env` uses `isRequire` under the hood: https://github.com/babel/babel/blob/74ed698c2e2140c0ef685b1542ef6ad8f20f0b20/packages/babel-preset-env/src/index.js#L288

It seems to me a bug of `@babel/helper-compilation-targets` if it didn't accept the `targets` passed to `preset-env`. Please file an issue to Babel if you can isolate the bug. Thanks!